src/redux/charts.ts

```diff
@@ -0,0 +1,235 @@
+
+  /**
+   * Whether or not the first store is fetching charts.
+   */
+  fetchingCharts: boolean;
+
+  /**
+   * Whether or not the second store is fetching charts.
+   */
+  fetchingCharts2: boolean;
+}
+
+/**
+ * An action for the `charts` reducer.
+ */
+type ChartsAction =
+  | ChartsUpdateChartTypeAction
+  | ChartsUpdateAction
+  | ChartsUpdateUsdDayAction
+  | ChartsUpdateUsdMonthAction;
+
+/**
+ * The action for updating the current type of chart loaded.
+ */
+interface ChartsUpdateChartTypeAction {
+  type: typeof CHARTS_UPDATE_CHART_TYPE;
+  payload: ChartType;
+  secondStore: boolean;
+}
+
+/**
+ * The action for updating chart data.
+ */
+interface ChartsUpdateAction {
+  type: typeof CHARTS_UPDATE;
+  payload: ChartsState['charts'];
+}
+
+/**
+ * The action for updating the ETH-USD day chart data.
+ */
+interface ChartsUpdateUsdDayAction {
+  type: typeof CHARTS_UPDATE_USD_DAY;
+  payload: ChartsState['chartsEthUSDDay'];
+}
+
+/**
+ * The action for updating the ETH-USD month chart data.
+ */
+interface ChartsUpdateUsdMonthAction {
+  type: typeof CHARTS_UPDATE_USD_MONTH;
+  payload: ChartsState['chartsEthUSDMonth'];
+}
+
+/**
+ * A message received from a chart data provider.
+ */
+interface ChartsReceivedMessage {
+  meta?: {
+    charts_type?: ChartType;
+    currency?: string;
+  };
+  payload: {
+    charts: {
+      [key: string]: [number, number][];
+    };
+  };
+}
+
+/**
+ * Updates the current chart type in state.
+ *
+ * @param chartType The new chart type.
+ * @param secondStore Whether or not this change should update the second chart
+ * store.
+ */
+export const chartsUpdateChartType = (
+  chartType: ChartType,
+  secondStore: boolean
+) => (dispatch: Dispatch<ChartsUpdateChartTypeAction>) =>
+  dispatch({
+    payload: chartType,
+    secondStore,
+    type: CHARTS_UPDATE_CHART_TYPE,
+  });
+
+/**
+ * Updates charts in state to reflect new data.
+ *
+ * @param message The `ChartsReceivedMessage`.
+ */
+export const assetChartsReceived = (message: ChartsReceivedMessage) => (
+  dispatch: Dispatch<
+    ChartsUpdateAction | ChartsUpdateUsdDayAction | ChartsUpdateUsdMonthAction
+  >,
+  getState: AppGetState
+) => {
+  const chartType = get(message, 'meta.charts_type');
+  const { charts: existingCharts } = getState().charts;
+  const assetCharts = get(message, 'payload.charts', {});
+  const { nativeCurrency } = getState().settings;
+  if (toLower(nativeCurrency) === message?.meta?.currency) {
+    const newChartData = mapValues(assetCharts, (chartData, address) => ({
+      ...existingCharts[address],
+      // .slice to prevent mutation
+      [chartType]: reverse(chartData?.slice()),
+    }));
+    const updatedCharts = {
+      ...existingCharts,
+      ...newChartData,
+    };
+    dispatch({
+      payload: updatedCharts,
+      type: CHARTS_UPDATE,
+    });
+  }
+
+  if (
+    message?.meta?.currency === currenyTypes.usd &&
+    assetCharts[ETH_ADDRESS]
+  ) {
+    if (message?.meta?.charts_type === 'm') {
```

Comment:
we can use `ChartTypes` for `m` and `d` now (line 178 & 183)

Reply:
@jinchung Okay, I updated this to use `ChartTypes`, thanks!